support/mock/mockrt.go

```diff
@@ -0,0 +1,326 @@
+	data, found := rt.store[c]
+	if found {
+		err := o.UnmarshalCBOR(bytes.NewReader(data))
+		if err != nil {
+			rt.Abort(exitcode.SysErrSerialization, err.Error())
+		}
+	}
+	return found
+}
+
+func (rt *Runtime) IpldPut(o runtime.CBORMarshaler) cid.Cid {
+	r := bytes.Buffer{}
+	err := o.MarshalCBOR(&r)
+	if err != nil {
+		rt.Abort(exitcode.SysErrSerialization, err.Error())
+	}
+	data := r.Bytes()
+	key, err := cidBuilder.Sum(data)
+	rt.store[key] = data
+	return key
+}
+
+func (rt *Runtime) Send(toAddr addr.Address, methodNum abi.MethodNum, params abi.MethodParams, value abi.TokenAmount) (runtime.SendReturn, exitcode.ExitCode) {
+	if rt.inTransaction {
+		rt.Abort(exitcode.SysErrorIllegalActor, "side-effect within transaction")
+	}
+	panic("implement me")
+}
+
+func (rt *Runtime) NewActorAddress() addr.Address {
+	panic("implement me")
+}
+
+func (rt *Runtime) CreateActor(codeId cid.Cid, address addr.Address) {
+	if rt.inTransaction {
+		rt.Abort(exitcode.SysErrorIllegalActor, "side-effect within transaction")
+	}
+	panic("implement me")
+}
+
+func (rt *Runtime) DeleteActor(address addr.Address) {
+	if rt.inTransaction {
+		rt.Abort(exitcode.SysErrorIllegalActor, "side-effect within transaction")
+	}
+	panic("implement me")
+}
+
+func (rt *Runtime) CurrIndices() indices.Indices {
+	panic("implement me")
+}
+
+func (rt *Runtime) Abort(errExitCode exitcode.ExitCode, msg string, args ...interface{}) {
+	panic(abort{errExitCode, fmt.Sprintf(msg, args...)})
+}
+
+func (rt *Runtime) AbortStateMsg(msg string) {
+	rt.Abort(exitcode.ErrPlaceholder, msg)
+}
+
+func (rt *Runtime) Syscalls() runtime.Syscalls {
+	panic("implement me")
+}
+
+func (rt *Runtime) Context() context.Context {
+	return rt.ctx
+}
+
+func (rt *Runtime) StartSpan(name string) runtime.TraceSpan {
+	return &TraceSpan{}
+}
+
+///// State handle implementation /////
+
+func (rt *Runtime) Construct(f func() runtime.CBORMarshaler) {
+	if rt.state.Defined() {
+		rt.Abort(exitcode.SysErrorIllegalActor, "state already constructed")
+	}
+	st := f()
+	rt.state = rt.IpldPut(st)
+}
+
+func (rt *Runtime) Readonly(st runtime.CBORUnmarshaler) {
+	found := rt.IpldGet(rt.state, st)
+	if !found {
+		rt.Abort(exitcode.SysErrInternal, "actor state not found: %v", rt.state)
+	}
+}
+
+func (rt *Runtime) Transaction(st runtime.CBORer, f func() interface{}) interface{} {
+	if rt.inTransaction {
+		rt.Abort(exitcode.SysErrorIllegalActor, "nested transaction")
+	}
+	rt.Readonly(st)
+	rt.inTransaction = true
+	ret := f()
+	rt.state = rt.IpldPut(st)
+	rt.inTransaction = false
+	return ret
+}
+
+///// Trace span implementation /////
+
+type TraceSpan struct {
+}
+
+func (t TraceSpan) End() {
+	// no-op
+}
+
+type abort struct {
+	code exitcode.ExitCode
+	msg  string
+}
+
+///// Inspection facilities /////
+
+func (rt *Runtime) StateRoot() cid.Cid {
+	return rt.state
+}
+
+func (rt *Runtime) GetState(o runtime.CBORUnmarshaler) {
+	rt.IpldGet(rt.state, o)
+}
```

Comment:
Any reason to have this in addition to `Readonly`?

Reply:
Removed